mibidata/tiff.py

```diff
@@ -285,47 +288,47 @@ def _page_metadata(page, description):
         _DATETIME_FORMAT)
 
     # check version for backwards compatibility
-    _convert_from_previous_metadata_versions(description)
+    _convert_from_previous(description)
+
+    metadata = {}
+    for key, val in description.items():
+        if key.startswith('mibi.'):
+            metadata[key[5:]] = val
+        elif key not in RESERVED_MIBITIFF_ATTRIBUTES:
+            metadata[key] = val
 
-    # retrieve required metadata
-    metadata = {key.split('.')[1]: value for key, value in
-                description.items() if key.startswith('mibi.')}
     metadata.update({
         'coordinates': (
             _cm_to_micron(page.tags['x_position'].value),
             _cm_to_micron(page.tags['y_position'].value)),
         'date': date,
         'size': size})
-    # retrieve user-defined metadata
-    try:
-        user_defined_attrs = description['user_defined_attributes']
-        print(user_defined_attrs)
-        metadata.update({key: value for key, value in description.items()
-                         if key in user_defined_attrs})
-    except KeyError:
-        # no user-defined metadata found
-        pass
 
     return metadata
 
 
-def _convert_from_previous_metadata_versions(description):
-    """Convert old metadata format to new one.
+def _convert_from_previous(description):
+    """Convert old metadata format for backwards compatibility.
 
-    This function ensures backwards compatibility for previous versions.
+    Most of these conversions would happen during MibiImage construction,
+    but we do them here in case reading the info only.
```

Comment:
This docstring is misleading. This function is necessary for reading an old TIFF file (written with previous versions of the TIFF creator and the MibiImage class) and converting its metadata to the new format.

Reply:
If reading a TIFF into a MibiImage, everything after that TODO is unnecessary and can be deleted, because the old metadata gets passed to the MibiImage constructor which does the conversion. My first draft relied on that, but then I realized (or rather a test caught) that then the `tiff.info` function returns unconverted data so this was my workaround to that.